lib/ch/query.ex

```diff
@@ -185,9 +185,10 @@ defimpl DBConnection.Query, for: Ch.Query do
   defp encode_param(n) when is_integer(n), do: Integer.to_string(n)
   defp encode_param(f) when is_float(f), do: Float.to_string(f)
   defp encode_param(b) when is_binary(b), do: b
-  defp encode_param(b) when is_boolean(b), do: b
+  defp encode_param(true), do: "1"
+  defp encode_param(false), do: "0"
   defp encode_param(%Decimal{} = d), do: Decimal.to_string(d, :normal)
-  defp encode_param(%Date{} = date), do: date
+  defp encode_param(%Date{} = date), do: Date.to_iso8601(date)
```

Comment:
I think this is the only required change.

Reply:
I pushed that but one test will still fail

Reply:
this is an issue because then you encode integers & floats and thus the ring test will fail for instance

Reply:
```elixir
defp encode_array_param(s) when is_binary(s) do
  [?', to_iodata(s, 0, s, []), ?']
end

defp encode_array_param(v) do 
  case encode_param(v) do
    b when is_binary(b) -> [?', b, ?']
    e -> e
  end
end
```

Reply:
Sure, but I'll need some time to check if a simpler approach is possible later today. I don't like that we test for value type in two places.